elements/rh-secondary-nav/rh-secondary-nav-menu.ts

```diff
@@ -0,0 +1,163 @@
+ *
+ * @csspart full-width    - {HTMLElement} container - The wrapper for full-width menus
+ * @csspart fixed-width   - {HTMLElement} container - The wrapper for fixed-width menus
+ * @csspart sections      - {HTMLElement} container - The wrapper for menu sections
+ * @csspart cta           - {HTMLElement} container - The wrapper for cta
+ */
+@customElement('rh-secondary-nav-menu') @pfelement()
+export class RhSecondaryNavMenu extends LitElement {
+  static readonly styles = [styles];
+
+  // TODO: change to variant instead of type which is used by html element properties such as input.
+  @property({ reflect: true }) type: 'fixed-width' | 'full-width' = 'full-width';
+
+  @queryAssignedNodes('cta', true)
+  private _ctaNodes: NodeListOf<HTMLElement> | undefined;
+
+  @queryAssignedNodes('section', true)
+  private _sectionsNodes: NodeListOf<HTMLElement> | undefined;
+
+  @query('#sections') _sections: HTMLElement | undefined;
+
+  @query('#cta') _cta: HTMLElement | undefined;
+
+  @query('#container') _container: HTMLElement | undefined;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.id ||= getRandomId('rh-secondary-nav-menu');
+  }
+
+  /**
+   * `isMobile` property is true when viewport `(min-width: ${tabletLandscapeBreakpoint})`.
+   * Property is observed for changes, and its value is updated using matchMediaControler
+   * when viewport changes at breakpoint or first load of the component.
+   */
+  @observed
+  @property({ reflect: false }) _isMobile = false;
+
+  /**
+   * Checks if passed in element is a RhSecondaryNavDropdown
+   * @param element:
+   * @returns {boolean}
+   */
+  static isDropdown(element: Element|null): element is RhSecondaryNavDropdown {
+    return element instanceof RhSecondaryNavDropdown;
+  }
+
+  /**
+   * Updates isMobile property with matchMedia when viewport changes
+   */
+  protected matchMedia = new MatchMediaController(this, `(min-width: ${tabletLandscapeBreakpoint})`, {
+    onChange: ({ matches }) => {
+      this._isMobile = !matches;
+    }
+  });
+
+  firstUpdated() {
+    this._isMobile = (window.outerWidth < parseInt(tabletLandscapeBreakpoint.toString().split('px')[0]));
+  }
+
+  render() {
+    const classes = { 'is-mobile': this._isMobile };
+
+    let menuContents;
+    if (this.type === 'full-width') {
+      menuContents = this.#fullWidthMenu();
+    } else {
+      menuContents = this.#fixedWidthMenu();
+    }
+    return html`<div id="container" class="${classMap(classes)}">${menuContents}</div>`;
+  }
+
+  /**
+   * Toggles visibility of menu by added or removing a `visible` attribute to the host
+   * @param visible {boolean}
+   */
+  toggleVisibility(visible: boolean) {
+    if (visible) {
+      this._container?.classList.add('visible');
+    } else {
+      this._container?.classList.remove('visible');
+    }
+  }
+
+  #fullWidthMenu() {
+    return html`
+      <div id="full-width" part="full-width">
+        <slot name="base">
```

Comment:
see prev re `base`

Reply:
Fixed in dae47d21f1dbb493c0ea02097e84efb399fe5e64